### ISSUE TITLE
[Informative] World Corruption/Player Data Corruption

### ISSUE THREAD

**Issue by Coraline77447** [MarkusBordihn/BOs-Player-Companions#6]
Hi, I want to let you know that one of my 1.18.2 worlds called C.W.G 2 (Cora World Guns 2) might have gotten corrupted after updating some mods including Player Companions which means that I've lost that world since it was made before Player Companions 1.18.2 1.0.0...
I'm afraid to load back into my other 1.18.2 world called Cora World which was also made before Player Companions 1.0.0 since that world might also get corrupted if I load back into it with it's mod list...

Feels like once I create a new world I can't update any mods anymore without causing potential world corruption or issues which sucks since it's important to always keep all mods updated to their latest version...

Java edition really does need full mod support like bedrock edition so that everyone won't have any potential problems or world corruption when updating mods to their latest version...

Btw I looked in the latest log before I made a test world for connectivity reasons since in the logs the game was sayin that some of the mods including Player Companions isn't using the vanilla connections and the error is about channels which apparently Player Comapnions isn't using the vanilla channel(s)...

Here, have a look see for yourself;

[2022-05-05-1.log](https://github.com/MarkusBordihn/BOs-Player-Companions/files/8631563/2022-05-05-1.log) 

I could go make a similar report to this one to the other mods that're affected by this problem but I wanted to inform you about the problem first since you're the most helpful and talkative mod creator I've met which I kind of consider you as a friend (or "online friend") since I'm friends with the dev team of Dungeons Mobs and Dungeons Gear since they're practically the same way* even tho I haven't been active on discord as often as I should...

*Unlike other mod creators like Corail31 (mod creator for Corail Recycler and Corail Tombstone) and Harmmond or whatever his username is (mod creator for Project MMO) which aren't as nice nor as respectful......

Why I say that Corail31 ain't nice or respectful is because Corail blocked me on Curseforge for playing with a mod that was made illegally (copyright) which was a knockoff of Corail Recycler back in 1.16.5 which at the time I didn't know that I was playing with an illegally made mod that violated copyright laws, what started it all was that I'd thought that Corail owned that mod and had updated it but I looked at the notification on courseforge again and realized that it was for Corail Recycler as the real legitimate one and had commented in about it and Corail blocks me with no warning and reason except that I assumed that it was because of the knockoff mod....

As for Harmmond I used to play with Ptoject MMO but after a while I got bored of the mod and that mod felt repetitive but before that I used to message on his discord server but what made me stay away from his discord server is cause of his rules like one of them is the most ridiculous rule which is no replying to anybody unless it was an absolutely important or an emergency of some kind but if it's not then he'll give ya a warning and if he gives ya three warnings then either you're getting muted or worse, banned from his discord server....

So I've kinda learned from that experience which is to not reply to people in general or on discord before sending a reply is to turn off the notification for said reply to somebody or not to reply to anyone unless it's important.....

UPDATE: So I'm back onto C.W.G 2 right now but I really, really, really, reaaaalllllly don't know why I wasn't able to join the world earlier today which I'm at a complete lost in my thoughts about this but at least I'm still able to join back onto C.W.G.2! Really didn't want to start all over but I'm afraid that I'm no longer updating any mods (including Player Companions) for C.W.G 2's mod list to avoid any potential issues and world corruption.........

**Comment by Coraline77447** [comment]
As I'd suspected but like I've said that I've managed to get back onto that world but I don't know exactly how long it has left before corruption finally does catch up to it, for both of my 1.18.2 worlds and my 1.18.1 world (C.W.G) is redacted, meaning that I'm keeping that world but I no longer play on it since I've moved onto 1.18.2 and eventually when 1.19 releases later on this year...
Btw when 1.19 comes I'll be playing more vanilla than modded for performance reasons since my ASUS VivoBook 15 ain't really cuttin the musterd just like my old Lenovo Ideapad 330s when playing with 50+ mods (but no more than 70+ mods however)...

Also like I've said that I'm no longer updating any mods that adds content except for core/library mods like Architectury...
Speaking of core/library mods, I hope that Player Companions does't acquire any dependencies since I have enough mods that require their own dependencies as is, I don't want any more core/lib mods clogging up my mods folder and using more resources for both the game and my computer...
Which that's partially the reason why I'll be playing more vanilla in 1.19+ because most of the mods I play with requires core/lib mods now which I understand that it makes mod development better but for some people like me however who has a shitty pc and can't get cash to afford a better pc would see that as a bad thing cause that's more mods to load and more resources for both the game as well as the computer to consume/use which I need as much framerates as I can possibly get so that my gaming experience won't be crappy with little to no frames at all....

**Comment by MarkusBordihn** [comment]
Yeah I know the pain with mod updates which broke your world. I had a lot of fun fixing stuff for my server after removing some mods from my mod pack.

But basically there is a specific format for the version number, which tells you if there will be a broken change or such a bug-fix which should not change anything. It works the following way:

- **1**.0.0 to **2**.0.0 means broken changes and new content
- **1.0**.0 to **1.1**.0 means new content without broken changes
- **1.0.0** to **1.0.1** means bug fixes

So everything with an increasing number at the beginning will very likely break something.

Technically libraries helping to avoid duplicating the code several time and also using less resource as they are implement directly. The library itself is normally not doing anything when it is not being used, it just like a place holder.
There are some libraries which contain more stuff as needed but in general as long the code is not used it will not cause any additional load.

The other reason why so maybe people using libraries is to avoid mod re-posting like https://stopmodreposts.org/
Because of the library which will be updated regular by the owner the re-poster have more work for keeping their version up to date together with the library. So they will avoid such mods and using other mods without any library.

**Comment by Coraline77447** [comment]
"Optimize the world including cache (to remove older artifacts)"
"But I may try to optimize both worlds and anything else that you've suggested for me to do to fix the problem"

Well I think that optimizing my two original worlds that I still have on a flash drive from 2016 (originally created in 1.10.2, long before Player Companions was created) will ever work the same as Cora World 1.18.2, C.W.G 2, and Cora Tinkers' since those two worlds that are 5+ years old might have a bit more than some corrupted data by now and at the time when Coraline World 2 was made (A survival world, since I didn't start playing modded until 1.12.2) I had overpopulated almost all of the villages, like one had nearly 80 villagers!
Nowadays if I copy Coraline World 2 from my flash drive to my saves folder, optimize, and load into it that without going too close to any of those villages that the performance is fine but as soon as I get to close or get within one of the villages that the lag and lag spikes emerges and stays with me till I log out and delete that world since I only have those two worlds for nostalgia purposes nowadays if I wanna feel nostalgic and wanna see those two worlds again.
Also last time I had saved and quitted that world 5+ years ago I was in my *mansion which was built within one of the overpopulated villages...

I know that this reply is sorta off topic but I was talking with myself about Minecraft world optimization and the fact that the older a world gets the chances that it'll be prone to have issues and performance issues and those two worlds that I've said as a story about them are prefect examples and they may not truly be savable since they're so old, unoptimized and overpopulated within those villages. 
Apparently at the time I'd thought that it was a good idea to convert every village that I've found into either a town or a city and boost up the population big time but that was back in 1.10.2/1.12.2. (lol)
If I try to reuse those world seeds from 5+ years ago in 1.14+ that those worlds won't generate the same way as they used too..

*I had copied my mansion off from one of the minecraft books from what was once my local library and I'm glad that I had took photos of it and a white wool outlining of the built for future references and if I really want to build it all over again. It was a big mansion that I had built multiple rooms that could house as many as 5 or 6 players.
Prefect for LAN or a multiplayer server with friends but sadly I don't have much friends who owns java edition as well as a pc.....
